src/components/PDFThumbnail/index.tsx

```diff
@@ -26,18 +32,40 @@ function PDFThumbnail({previewSourceURL, style, isAuthTokenRequired = false, ena
                 }}
                 externalLinkTarget="_blank"
                 onPassword={onPassword}
+                onLoad={() => {
+                    setHasError(false);
+                }}
+                onLoadError={() => {
+                    if (onLoadError) {
+                        onLoadError();
+                    }
+                    setHasError(true);
+                }}
+                error={() => null}
             >
                 <View pointerEvents="none">
                     <Thumbnail pageIndex={0} />
                 </View>
             </Document>
         ),
-        [isAuthTokenRequired, previewSourceURL, onPassword],
+        [isAuthTokenRequired, previewSourceURL, onPassword, onLoadError],
     );
 
     return (
-        <View style={[style, styles.overflowHidden]}>
-            <View style={[styles.w100, styles.h100, styles.alignItemsCenter, styles.justifyContentCenter]}>{enabled && thumbnail}</View>
+        <View style={[style, styles.h100, styles.overflowHidden]}>
+            <View style={[styles.w100, styles.h100, !hasError && styles.alignItemsCenter, !hasError && styles.justifyContentCenter]}>
```

Comment:
Why are conditional `alignItemsCenter` and `justifyContentCenter `styling passed here? 

Instead of checking hasError twice for both stylings, please merge them.
```diff
- !hasError && styles.alignItemsCenter, !hasError && styles.justifyContentCenter
+ !hasError && {...styles.alignItemsCenter, ...styles.justifyContentCenter}
```